File: server/app/schemas/invite.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class InviteCreateRequest(BaseModel):
    max_uses: int | None = Field(default=None, ge=1)
    expires_in_hours: int | None = Field(default=None, ge=1)


class InviteRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    code: str
    created_by_id: uuid.UUID
    max_uses: int | None
    uses_count: int
    expires_at: datetime | None
    created_at: datetime

File: app/lib/models/invite.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'invite.freezed.dart';
part 'invite.g.dart';

@freezed
abstract class Invite with _$Invite {
  const factory Invite({
    required String id,
    required String code,
    required String createdById,
    required int? maxUses,
    required int usesCount,
    required DateTime? expiresAt,
    required DateTime createdAt,
  }) = _Invite;

  factory Invite.fromJson(Map<String, dynamic> json) => _$InviteFromJson(json);
}
